### PR TITLE
Add JRuby 9.3 and 9.4 to the test matrix under both Bazel versions.

### PR DESCRIPTION
**Pull request by JasonLunn**
Prepare for 9.2 to be removed.

### CODE REVIEW

File: .github/workflows/test_ruby.yml [hunk 1]
```diff
@@ -21,10 +21,13 @@ jobs:
...
-          - { name: JRuby 9.2, ruby: jruby-9.2.20.1, bazel: 5.1.1}
-          - { name: JRuby 9.3, ruby: jruby-9.3.4.0, bazel: 5.1.1}
+          - { name: JRuby 9.2, ruby: jruby-9.2.20.1, bazel: 5.1.1} # To be removed
+          - { name: JRuby 9.3, ruby: jruby-9.3.10.0, bazel: 5.1.1}
+          - { name: JRuby 9.4, ruby: jruby-9.4.3.0, bazel: 5.1.1}
           - { name: Ruby 2.7 (Bazel6), ruby: ruby-2.7.0, bazel: 6.0.0}
-          - { name: JRuby 9.2 (Bazel6), ruby: jruby-9.2.20.1, bazel: 6.0.0}
+          - { name: JRuby 9.2 (Bazel6), ruby: jruby-9.2.20.1, bazel: 6.0.0} # To be removed
+          - { name: JRuby 9.3 (Bazel6), ruby: jruby-9.3.10.0, bazel: 6.0.0}
+          - { name: JRuby 9.4 (Bazel6), ruby: jruby-9.4.3.0, bazel: 6.0.0}
```
Comment (mkruskal-google) [inline .github/workflows/test_ruby.yml:30]: This was just a test of Bazel 6, you can just update it to 9.4 and not create 2 new ones.  Note: this *needs* to come with changes to our copybara config in google3.  We could add them after import, or you could just create a cl

File: .github/workflows/test_ruby.yml [hunk 2]
```diff
@@ -21,10 +21,13 @@ jobs:
           - { name: Ruby 3.0, ruby: ruby-3.0.2, bazel: 5.1.1}
           - { name: Ruby 3.1, ruby: ruby-3.1.0, bazel: 5.1.1}
           - { name: Ruby 3.2, ruby: ruby-3.2.0, bazel: 5.1.1}
-          - { name: JRuby 9.2, ruby: jruby-9.2.20.1, bazel: 5.1.1}
-          - { name: JRuby 9.3, ruby: jruby-9.3.4.0, bazel: 5.1.1}
+          - { name: JRuby 9.2, ruby: jruby-9.2.20.1, bazel: 5.1.1} # To be removed
+          - { name: JRuby 9.3, ruby: jruby-9.3.10.0, bazel: 5.1.1}
```
Comment (mkruskal-google) [inline .github/workflows/test_ruby.yml:25]: 9.3 is also likely to be removed

File: .github/workflows/test_ruby.yml [hunk 3]
```diff
@@ -21,10 +21,13 @@ jobs:
           - { name: Ruby 3.0, ruby: ruby-3.0.2, bazel: 5.1.1}
           - { name: Ruby 3.1, ruby: ruby-3.1.0, bazel: 5.1.1}
           - { name: Ruby 3.2, ruby: ruby-3.2.0, bazel: 5.1.1}
-          - { name: JRuby 9.2, ruby: jruby-9.2.20.1, bazel: 5.1.1}
-          - { name: JRuby 9.3, ruby: jruby-9.3.4.0, bazel: 5.1.1}
+          - { name: JRuby 9.2, ruby: jruby-9.2.20.1, bazel: 5.1.1} # To be removed
```
Comment (mkruskal-google) [inline .github/workflows/test_ruby.yml:24]: Add a TODO with a bug?

File: .github/workflows/test_ruby.yml [hunk 4]
```diff
@@ -112,10 +115,13 @@ jobs:
...
+          - { name: JRuby 9.3, ruby: jruby-9.3.10.0, bazel: 5.1.1}
+          - { name: JRuby 9.4, ruby: jruby-9.4.3.0, bazel: 5.1.1}
           - { name: Ruby 2.7 (Bazel6), ruby: ruby-2.7.0, bazel: 6.0.0}
-          - { name: JRuby 9.2 (Bazel6), ruby: jruby-9.2.20.1, bazel: 6.0.0}
+          - { name: JRuby 9.2 (Bazel6), ruby: jruby-9.2.20.1, bazel: 6.0.0}  # To be removed
+          - { name: JRuby 9.3 (Bazel6), ruby: jruby-9.3.10.0, bazel: 6.0.0}
+          - { name: JRuby 9.4 (Bazel6), ruby: jruby-9.4.3.0, bazel: 6.0.0}
```
Comment (mkruskal-google) [inline .github/workflows/test_ruby.yml:124]: Same thing here, we don't need 3 of these

File: .github/workflows/test_ruby.yml (hunk 3 above)
Comment (JasonLunn) [reply .github/workflows/test_ruby.yml]: Removed the comment, will track this todo offline.

File: .github/workflows/test_ruby.yml (hunk 1 above)
Comment (JasonLunn) [reply .github/workflows/test_ruby.yml]: Done

File: .github/workflows/test_ruby.yml (hunk 4 above)
Comment (JasonLunn) [reply .github/workflows/test_ruby.yml]: Done

**Review by haberman — APPROVED** [review]

**Review by fowles — APPROVED** [review]

**Review by zhangskz — APPROVED** [review]

**Pull request closed without merging**